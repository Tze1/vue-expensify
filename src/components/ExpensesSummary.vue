<template>
  <div class="expensessummary component">
    <p>
      <span class="expensessummary-showing">Showing&nbsp;</span>
      <span class="expensessummary-count">{{ expensesCount }}</span>&nbsp;expense(s)
      <span v-if="expensesFiltered" class="expensessummary-filtered">(filtered)&nbsp;</span>&nbsp;totalling
      <span class="expensessummary-amount">{{ expensesSum }}</span>
    </p>
  </div>
</template>

<script>
  import { mapState } from 'vuex';
  import numeral from 'numeral';

  export default {
    name: 'ExpensesSummary',
    computed: mapState({
      expensesFiltered () {
        return this.$store.getters.expensesFiltered;
      },
      expensesCount () {
        return this.$store.getters.filteredCount;
      },
      expensesSum () {
        return numeral(this.$store.getters.filteredSum / 100).format('$0,0.00');
      },
    }),
  };
</script>

<style scoped lang="scss">
  @import '../styles/app.scss';

  .expensessummary.component {
    > p {
      font-size: $font-size-sm;
      margin-bottom: $space-sm;
      margin-top: $space-xs;

      .expensessummary-showing {
        display: none;

        @include mq("tablet") {
          display: inline;
        }
      }

      .expensessummary-count,
      .expensessummary-amount {
        font-size: $font-size-default;
        font-weight: bold;
      }
    }
  }
</style>
