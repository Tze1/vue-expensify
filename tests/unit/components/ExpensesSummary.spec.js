import { shallowMount } from '@vue/test-utils';
import numeral from 'numeral';
import ExpensesSummary from '@/components/ExpensesSummary';

describe('ExpensesSummary', () => {
  const $store = {
    getters: {
      expensesFiltered () {
        return this.$store.getters.expensesFiltered;
      },
      expensesCount () {
        return this.$store.getters.filteredCount;
      },
      expensesSum () {
        return numeral(this.$store.getters.filteredSum / 100).format('$0,0.00');
      },
    },
  };
  const wrapper = shallowMount(ExpensesSummary, {
    mocks: {
      $store,
    },
  });

  it('renders ExpensesSummary component properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
