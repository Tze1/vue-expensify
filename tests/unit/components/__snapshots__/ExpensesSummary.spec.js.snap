// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpensesSummary renders ExpensesSummary component properly 1`] = `
<div class="expensessummary component">
  <p><span class="expensessummary-count"></span>&nbsp;expense(s)&nbsp;
    <!---->totalling&nbsp;
    <span class="expensessummary-amount">$0.00</span>
    <!---->
  </p>
</div>
`;
